lfs/batcher.go

```diff
@@ -51,16 +55,22 @@ func (b *Batcher) acceptInput() {
 	exit := false
 
 	for {
-		batch := b.newBatch()
+		batch := make([]Transferable, 0, b.batchSize)
 	Loop:
 		for len(batch) < b.batchSize {
-			t, ok := <-b.input
-			if !ok {
-				exit = true // input channel was closed by Exit()
-				break Loop
+			select {
+			case t, ok := <-b.input:
+				if ok {
+					batch = append(batch, t)
+				} else {
+					exit = true // input channel was closed by Exit()
+					break Loop
+				}
+			case <-time.After(time.Millisecond * 300):
```

Comment:
This timeout tracks how long it takes to iterate through a `[]*lfs.WrappedPointer` from the push/pre-push/fetch commands, correct? Is it hanging if some of the pointers are skipped or erroring for some reason?